apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -120,16 +120,17 @@ export default class CoreLibrary {
     // For the calculations below, keep in mind that x and y are located at the horizontal center and the top of the sprite, respectively.
     // In other words, x and y indicate the default position of the bubble's triangular tip.
     y = Math.min(y, APP_HEIGHT);
+    let originalX = x;
```

Comment:
@ajpal 

**Bug:** We were checking speech bubble "overflow" on both left/right edges with the same x value AND potentially changing that value between the two checks. Therefore, the second check wasn't handled accurately if the bubble was larger than the full app width.

![2021-11-18 08-40-24 2021-11-18 08_41_07](https://user-images.githubusercontent.com/43474485/142429188-1522f99a-37b7-49eb-ab50-cbffc1b48e9e.gif)

**Fix:** Temporarily store the original X value and use that for both checks.
![2021-11-18 08-41-59 2021-11-18 08_42_36](https://user-images.githubusercontent.com/43474485/142429254-128a1db2-a4ed-48ff-b7df-a3d375fb3c8e.gif)